ctriface/iface.go

```diff
@@ -205,17 +208,18 @@ func (o *Orchestrator) StartVMWithEnvironment(ctx context.Context, vmID, imageNa
 	if o.GetUPFEnabled() {
 		logger.Debug("Registering VM with the memory manager")
 
+		logger.Debugf("TEST (startWithEnv): current vmID used to registerVM is %v", vmID)
 		stateCfg := manager.SnapshotStateCfg{
-			VMID:           vmID,
-			GuestMemPath:   o.getMemoryFile(vmID),
-			BaseDir:        o.getVMBaseDir(vmID),
-			GuestMemSize:   int(conf.MachineCfg.MemSizeMib) * 1024 * 1024,
-			IsLazyMode:     o.isLazyMode,
-			VMMStatePath:   o.getSnapshotFile(vmID),
-			WorkingSetPath: o.getWorkingSetFile(vmID),
-			// FIXME (gh-807)
-			//InstanceSockAddr: resp.UPFSockPath,
+			VMID:             vmID,
+			GuestMemPath:     o.getMemoryFile(vmID),
+			BaseDir:          o.getVMBaseDir(vmID),
+			GuestMemSize:     int(conf.MachineCfg.MemSizeMib) * 1024 * 1024,
+			IsLazyMode:       o.isLazyMode,
+			VMMStatePath:     o.getSnapshotFile(vmID),
+			WorkingSetPath:   o.getWorkingSetFile(vmID),
+			InstanceSockAddr: o.uffdSockAddr,
 		}
+		logger.Debugf("TEST: show to-reg snapStat: %+v", stateCfg)
 		if err := o.memoryManager.RegisterVM(stateCfg); err != nil {
```

Comment:
ok, VM registration happens here. You should not reuse/repurpose this interface for LoadSnapshot. If needed, create a new interface.

so we would have 2 interfaces for registration: one for the original VMs and one fo the VMs which are loaded from a snap.